web/src/components/applications-page/encrypt-secret-drawer/index.tsx

```diff
@@ -0,0 +1,216 @@
+import {
+  Box,
+  Button,
+  Checkbox,
+  Drawer,
+  FormControl,
+  FormControlLabel,
+  InputLabel,
+  MenuItem,
+  Select,
+  TextField,
+  Typography,
+} from "@mui/material";
+import { useFormik } from "formik";
+import { FC, memo, useCallback, useEffect, useRef, useMemo } from "react";
+import * as yup from "yup";
+import { TextWithCopyButton } from "~/components/text-with-copy-button";
+import { UI_TEXT_CANCEL } from "~/constants/ui-text";
+import { useAppDispatch, useAppSelector } from "~/hooks/redux";
+import { Piped, fetchPipeds, selectAllPipeds } from "~/modules/pipeds";
+import { sortFunc } from "~/utils/common";
+import {
+  clearSealedSecret,
+  generateSealedSecret,
+} from "~/modules/sealed-secret";
+
+export interface EncryptSecretDrawerProps {
+  open: boolean;
+  onClose: () => void;
+}
+
+const DRAWER_TITLE = "Encrypting secret data";
+const BASE64_CHECKBOX_LABEL =
+  "Use base64 encoding before encrypting the secret";
+
+const validationSchema = yup.object({
+  secretData: yup.string().required(),
+  base64: yup.bool(),
+  pipedId: yup.string().required("Please select a piped"),
+});
+
+export const EncryptSecretDrawer: FC<EncryptSecretDrawerProps> = memo(
+  function EncryptSecretDrawer({ open, onClose }) {
+    const dispatch = useAppDispatch();
+
+    const pipeds = useAppSelector<Piped.AsObject[]>(selectAllPipeds);
+    const sealedSecret = useAppSelector<string | null>(
+      (state) => state.sealedSecret.data
+    );
+
+    const pipedOptions = useMemo(() => {
+      return pipeds
+        .filter((piped) => !piped.disabled)
+        .sort((a, b) => sortFunc(a.name, b.name));
+    }, [pipeds]);
+
+    useEffect(() => {
+      if (open) {
+        dispatch(fetchPipeds(false));
+      }
+    }, [dispatch, open]);
+
+    const formik = useFormik({
+      initialValues: {
+        secretData: "",
+        base64: false,
+        pipedId: "",
+      },
+      validationSchema,
+      async onSubmit(values) {
+        await dispatch(
+          generateSealedSecret({
+            data: values.secretData,
+            pipedId: values.pipedId,
+            base64Encoding: values.base64,
+          })
+        );
+      },
+    });
+
+    const handleClose = useCallback(() => {
+      onClose();
+      dispatch(clearSealedSecret());
+    }, [dispatch, onClose]);
+
+    const prevOpen = useRef(false);
+    useEffect(() => {
+      if (!prevOpen.current && open) {
+        formik.resetForm();
+      }
+      prevOpen.current = open;
+    }, [open, formik]);
```

Comment:
You can prevent effect infinite loop  by using { resetForm } dependency instead of formik which contain lots of infomation. It will make more sense rather than new ref created.


```suggestion
    const { resetForm } = formik;
    
    useEffect(() => {
      if (open) {
        resetForm();
      }
    }, [open, resetForm]);
```